### PR TITLE
#2151 Disable menus on empty selection

### LINKED ISSUE #2151: Visible menus on empty selectios
-  fix when validate menu and copy as menu is visible
-  fix copy as text on DRepresentarionDescriptor returned text
-  disable Migration and Open menus on empty selections
-  disable Validation menu if the selection from the action is disabled

### PR DESCRIPTION
**Pull request by georgiana-ecobici**
Resolves #2151

- Migrate, Open, Open and Show in Diagram Editor
- Disable Validate action when it contains empty selection

Change-Id: I6c7d64dcb5702a173a5bb5f5ef89df7e12e7e237
Signed-off-by: Georgiana Ecobici <georgiana-elena.ecobici@thalesgroup.com>

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.validation.ui/src/org/polarsys/capella/core/validation/ui/DynamicActionProvider.java [hunk 1]
```diff
@@ -105,6 +105,8 @@ public ValidateAction createDefaultValidation() {
     ISelection selection = selectionProvider.getSelection();
     if (!selection.isEmpty()) {
       validationAction.selectionChanged(new SelectionChangedEvent(selectionProvider, selection));
+    } else {
+      validationAction.setEnabled(false);
     }
```
Comment (georgiana-ecobici) [inline core/plugins/org.polarsys.capella.core.validation.ui/src/org/polarsys/capella/core/validation/ui/DynamicActionProvider.java:110]: This change is for the case when an element was previously selected, but you click on empty space in SB (selection inside the validate action is empty).

image

File: core/plugins/org.polarsys.capella.core.data.migration/plugin.xml [hunk 2]
```diff
@@ -90,6 +90,9 @@
                   commandId="org.polarsys.capella.core.data.migration.project">
                <visibleWhen
                      checkEnabled="true">
+		               <iterate
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.data.migration/plugin.xml:93]: image

**Review by pdulth — CHANGES_REQUESTED** [review]
We will merge it for IT2,
The first fix makes the migration project displayed but disabled when we click on any other elements.

**Comment by pdulth** [comment]
When you use one reference for the 3 commands, when clicking on an element, the non enabled will remain visible.

image

To have the same behavior than previous version, can you try to add 3 dedicated references, one per handler rather than one with the three handlers?
image
image

File: core/plugins/org.polarsys.capella.core.data.migration/src/org/polarsys/capella/core/data/migration/handlers/MigrationPropertyTester.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * 
+ */
+public class MigrationPropertyTester extends PropertyTester {
+
+  @Override
+  public boolean test(Object receiver, String property, Object[] args, Object expectedValue) {
+    // check if is project, aird or model migration
+    switch(property) {
+    case "isValidMigrationProject":
+      return new ProjectMigrationHandler().isValidSelection(Arrays.asList(receiver));
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.data.migration/src/org/polarsys/capella/core/data/migration/handlers/MigrationPropertyTester.java]: When we disable commands through "Window / Preferences / Capella / CommandConfigurability / Migration"
the handler on the command is replaced by another one by "PreferenceHandler.getInstance(command).activateHandler

If we use the "classical one" here, the command will be disabled and still visible even if we used a visibleWhen in the menu command.
In the new patchset, we retrieve the current command handler (classical one or PreferenceHelper disabling one) then look if it is enabled.

**Review by pdulth — APPROVED** [review]

**Pull request merged**